src/main/webapp/app/entities/result/result.component.ts

```diff
@@ -181,4 +276,23 @@ export class ResultComponent implements OnInit, OnChanges {
         }
         return ['far', 'times-circle'];
     }
+
+    /**
+     * Find latest result in results array
+     *
+     * @param {Result} results
+     */
+    getLatestResult(results: Result[]) {
+        results.sort((r1: Result, r2: Result) => {
```

Comment:
This mutates the original array and is considered a bad practice in JS if there is no performance indication.
You can e.g. use https://lodash.com/docs/4.17.15#sortBy.

The problem is when calling the method you would think that it just returns the latest result, while in reality it also mutates the given argument (`results).

I would propose a different solution here:
- One line
- Doesn't mutate arguments
- Easy to understand if familiar with functional programming
```
return arr.reduce((acc, result) => result.completionDate > acc.completionDate ? result : acc); 
```

Reply:
Thanks for the suggestion! I implemented it.